src/_premake_init.lua

```diff
@@ -1348,6 +1344,13 @@
 	end)
 
 
+	-- 31 January 2017
+
+	api.deprecateValue("flags", "C++11", 'Use `language "C++11"` instead', function(value) end, function(value) end)
+	api.deprecateValue("flags", "C++14", 'Use `language "C++14"` instead', function(value) end, function(value) end)
+	api.deprecateValue("flags", "C90",   'Use `language "C90"` instead',   function(value) end, function(value) end)
+	api.deprecateValue("flags", "C99",   'Use `language "C99"` instead',   function(value) end, function(value) end)
+	api.deprecateValue("flags", "C11",   'Use `language "C11"` instead',   function(value) end, function(value) end)
```

Comment:
You could just do:

```
language "C++14"
filter { "files:*.c" }
    language "C99"
```

but that said, if we went the "dialect" route, I would certainly prefer to actually use that name.
So either:
```
dialect {
    cpp = "C++14"
    c = "C90",
    d = "D3.4",  -- if such a thing was a thing.
}
```
or as suggested: 
```
cdialect 'C90'
cppdialect 'C++14'
ddialect 'D3.4"
```
I'm somewhat indifferent to the approach, I think both have merit..

You guys want me to prototype that instead??